spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/StructTypeImplicits.scala

```diff
@@ -417,4 +417,45 @@ object StructTypeImplicits {
     }
   }
 
+  implicit class StructTypeComparisonWhileIgnoringNullability(val schema: StructType) {
+    /**
+     * Wraps an array of StructField with StructType
+     *
+     * @param fieldSchema is the actual schema to be wrapped
+     * @return StructType containing the StructFields
+     */
+    def wrapStructFiledWithStructType(fieldSchema: Option[DataType]): Boolean = {
+      val st = StructType(Seq(StructField("errCol", fieldSchema.headOption.head)))
```

Comment:
Bear in mind that `"errCol"` i.e. name of the err column, can be parametrized as well, see `case class ErrorMessageArray(errorColumnName: String = ErrorMessageArray.defaultErrorColumnName)
`

So perhaps this class or function should also follow similar logic - making it parametrizable, maybe with some default if you want.